### PR TITLE
Add DecodedVector::equalValueAt API

### PR DESCRIPTION
**Pull request by duanmeng**
Details in https://github.com/facebookincubator/velox/pull/6390#discussion_r1317580319

### CODE REVIEW

**Comment by duanmeng** [comment]
@bikramSingh91 I add the `equalValueAt` API in this PR according to our discussion in https://github.com/facebookincubator/velox/pull/6390#discussion_r1317580319. This API would combine with the `isNullAt` API since a DecodedVector may have extra nulls (eg, Dictionary encoding). Could you help to take a look? cc @mbasmanova

File: velox/vector/DecodedVector.h [hunk 1]
```diff
@@ -209,6 +209,16 @@ class DecodedVector {
     return bits::isBitNull(nulls_, indices_[idx]);
   }
 
+  /// Return whether this top-level vector has the same value at the given index
+  /// as the other top-level vector at the other vector's index.
+  bool equalValueAt(
+      const DecodedVector& other,
+      vector_size_t idx,
```
Comment (mbasmanova) [inline velox/vector/DecodedVector.h:216]: nit: idx -> index

File: velox/vector/DecodedVector.h [hunk 2]
```diff
@@ -209,6 +209,16 @@ class DecodedVector {
...
+      vector_size_t otherIndex) const {
+    return baseVector_->equalValueAt(
```
Comment (mbasmanova) [inline velox/vector/DecodedVector.h:218]: I think we need to check for nulls (added on to of baseVector_), no?

Would it be possible to add some tests?

**Review by mbasmanova — COMMENTED** [review]
@laithsakka Laith, would you help review this PR?

File: velox/vector/DecodedVector.h (hunk 2 above)
Comment (mbasmanova) [inline velox/vector/DecodedVector.h:218]: Comparison of complex types requires to check for nested nulls and throw. I think we need to add CompareFlags parameter to this API to allow for that.

**Pull request closed without merging**